disnake/guild_scheduled_event.py

```diff
@@ -0,0 +1,372 @@
+
+        Returns
+        -------
+        :class:`GuildScheduledEvent`
+            The updated guild scheduled event instance.
+
+        Raises
+        ------
+        Forbidden
+            You do not have proper permissions to edit the event.
+        NotFound
+            The event does not exist.
+        HTTPException
+            Editing the event failed.
+        """
+
+        fields: Dict[str, Any] = {}
+
+        if privacy_level is not MISSING:
+            if not isinstance(privacy_level, StagePrivacyLevel):
+                raise ValueError("privacy_level must be an instance of StagePrivacyLevel")
+
+            fields["privacy_level"] = privacy_level.value
+
+        if entity_type is not MISSING:
+            if not isinstance(entity_type, GuildScheduledEventEntityType):
+                raise ValueError("entity_type must be an instance of GuildScheduledEventEntityType")
+
+            fields["entity_type"] = entity_type.value
+
+        if entity_metadata is not MISSING:
+            if not isinstance(entity_metadata, GuildScheduledEventMetadata):
+                raise ValueError(
+                    "entity_metadata must be an instance of GuildScheduledEventMetadata"
+                )
+
+            fields["entity_metadata"] = (
+                None if entity_metadata is None else entity_metadata.to_dict()
+            )
+
+        if status is not MISSING:
+            if not isinstance(status, GuildScheduledEventStatus):
+                raise ValueError("status must be an instance of GuildScheduledEventStatus")
+
+            fields["status"] = status.value
+
+        if name is not MISSING:
+            fields["name"] = name
+
+        if description is not MISSING:
+            fields["description"] = description
+
+        if channel_id is not MISSING:
+            fields["channel_id"] = channel_id
+
+        if scheduled_start_time is not MISSING:
+            fields["scheduled_start_time"] = scheduled_start_time.isoformat()
+
+        if scheduled_end_time is not MISSING:
+            fields["scheduled_end_time"] = scheduled_end_time.isoformat()
+
+        data = await self._state.http.edit_guild_scheduled_event(
+            guild_id=self.guild_id, event_id=self.id, **fields
+        )
+        return GuildScheduledEvent(state=self._state, data=data)
+
+    async def fetch_users(
+        self,
+        *,
+        limit: int = None,
+        with_members: bool = True,
+        before_id: int = None,
+        after_id: int = None,
+    ) -> List[Union[Member, User]]:
+        """|coro|
+
+        Get a list of guild scheduled event users subscribed to this guild scheduled event.
+
+        Parameters
+        ----------
+        limit: :class:`int`
+            How many users to receive from the event.
+        with_members: :class:`bool`
+            Whether to include some users as members. Defaults to ``True``.
+        before_id: :class:`int`
+            Consider only users before given user id.
```

Comment:
"before" means "earlier" in terms of snowflakes, since snowflakes are shifted timestamps.

No, it's not paginated, see [get_guild_scheduled_event_users](https://discord.com/developers/docs/resources/guild-scheduled-event#get-guild-scheduled-event-users), plus this [comment](https://github.com/discord/discord-api-docs/pull/3586#issuecomment-964582791)

Reply:
> since snowflakes are shifted timestamps
I know about that, but I was wondering if the user list is actually sorted by those user snowflakes/timestamps, or instead by e.g. the event subscription timestamp of each user.

Reply:
Official docs are saying that `before` and `after` both are user IDs, so it seems like they mean that the user list is sorted by IDs.
![image](https://user-images.githubusercontent.com/50338932/142696227-a48958d7-71e1-4ff8-9f3d-adcfe17ee169.png)

Reply:
I see.
Seems very similar to `Client.fetch_guilds` ([docs](https://discord.com/developers/docs/resources/user#get-current-user-guilds)), which is paginated and also receives chunks of 100 items with `before`/`after` parameters.